src/analyzer/plugins/analyzerqueenmarybeats.cpp

```diff
@@ -77,22 +77,21 @@ bool AnalyzerQueenMaryBeats::processSamples(const CSAMPLE* pIn, SINT iLen) {
 bool AnalyzerQueenMaryBeats::finalize() {
     m_helper.finalize();
 
-    int nonZeroCount = static_cast<int>(m_detectionResults.size());
+    std::size_t nonZeroCount = m_detectionResults.size();
     while (nonZeroCount > 0 && m_detectionResults.at(nonZeroCount - 1) <= 0.0) {
         --nonZeroCount;
     }
 
+    std::size_t required_size = std::max(static_cast<std::size_t>(0), nonZeroCount - 2);
+
     std::vector<double> df;
-    std::vector<double> beatPeriod;
-    const auto required_size = std::max(0, nonZeroCount - 2);
     df.reserve(required_size);
-    beatPeriod.reserve(required_size);
+    auto beatPeriod = std::vector<double>(required_size);
 
     // skip first 2 results as it might have detect noise as onset
     // that's how vamp does and seems works best this way
-    for (int i = 2; i < nonZeroCount; ++i) {
+    for (std::size_t i = 2; i < nonZeroCount; ++i) {
         df.push_back(m_detectionResults.at(i));
-        beatPeriod.push_back(0.0);
     }
```

Comment:
while this will cause `beatPeriod` to be a vector of `required_size` elements, that constructor only default constructs the elements, which in the case of double, will cause them to be **uninitialized**. The previous behavior explicitly zero-initialized them via the `push_back`. Is this safe? Can you add a comment that this subtle behavior is intentional (or just do `std::vector<double>(required_size, 0.0);` to initialize the contents)?